src/resonate.ts

```diff
@@ -81,14 +81,40 @@ export class Resonate {
     this.ttl = ttl;
     this.encoder = new JsonEncoder();
 
-    if (!url) {
+    // Determine the URL based on priority: url arg > RESONATE_URL > RESONATE_HOST+PORT
+    let resolvedUrl = url;
+    if (!resolvedUrl) {
+      if (process.env.RESONATE_URL) {
+        resolvedUrl = process.env.RESONATE_URL;
+      } else {
+        const resonateHost = process.env.RESONATE_HOST;
+        const resonatePort = process.env.RESONATE_PORT;
```

Comment:
Can we default this to 8001?

Also, do you think we need a scheme env variable that defaults to "http" or is it safe to assume that is part of the HOST?